DQM/EcalMonitorTasks/python/CollectionTags_cfi.py

```diff
@@ -45,16 +45,17 @@
     EEBasicCluster = cms.untracked.InputTag("particleFlowClusterECAL"),
     EBSuperCluster = cms.untracked.InputTag("particleFlowSuperClusterECAL", "particleFlowSuperClusterECALBarrel"),
     EESuperCluster = cms.untracked.InputTag("particleFlowSuperClusterECAL", "particleFlowSuperClusterECALEndcapWithPreshower"),
-    EBCpuDigi = cms.untracked.InputTag("ecalDigis@cpu", "ebDigis"),
-    EECpuDigi = cms.untracked.InputTag("ecalDigis@cpu", "eeDigis"),
-    EBGpuDigi = cms.untracked.InputTag("ecalDigis@cuda", "ebDigis"),
-    EEGpuDigi = cms.untracked.InputTag("ecalDigis@cuda", "eeDigis"),
-    EBCpuUncalibRecHit = cms.untracked.InputTag("ecalMultiFitUncalibRecHit@cpu", "EcalUncalibRecHitsEB"),
-    EECpuUncalibRecHit = cms.untracked.InputTag("ecalMultiFitUncalibRecHit@cpu", "EcalUncalibRecHitsEE"),
-    EBGpuUncalibRecHit = cms.untracked.InputTag("ecalMultiFitUncalibRecHit@cuda", "EcalUncalibRecHitsEB"),
-    EEGpuUncalibRecHit = cms.untracked.InputTag("ecalMultiFitUncalibRecHit@cuda", "EcalUncalibRecHitsEE"),
-    EBCpuRecHit = cms.untracked.InputTag("ecalRecHit@cpu", "EcalRecHitsEB"),
-    EECpuRecHit = cms.untracked.InputTag("ecalRecHit@cpu", "EcalRecHitsEE"),
-    EBGpuRecHit = cms.untracked.InputTag("ecalRecHit@cuda", "EcalRecHitsEB"),
-    EEGpuRecHit = cms.untracked.InputTag("ecalRecHit@cuda", "EcalRecHitsEE")
+    EBCpuDigi = cms.untracked.InputTag("hltEcalDigisLegacy", "ebDigis"),
+    EECpuDigi = cms.untracked.InputTag("hltEcalDigisLegacy", "eeDigis"),
+    EBGpuDigi = cms.untracked.InputTag("hltEcalDigisFromGPU", "ebDigis"),
+    EEGpuDigi = cms.untracked.InputTag("hltEcalDigisFromGPU", "eeDigis"),
+    EBCpuUncalibRecHit = cms.untracked.InputTag("hltEcalUncalibRecHitLegacy", "EcalUncalibRecHitsEB"),
+    EECpuUncalibRecHit = cms.untracked.InputTag("hltEcalUncalibRecHitLegacy", "EcalUncalibRecHitsEE"),
+    EBGpuUncalibRecHit = cms.untracked.InputTag("hltEcalUncalibRecHitFromSoA", "EcalUncalibRecHitsEB"),
+    EEGpuUncalibRecHit = cms.untracked.InputTag("hltEcalUncalibRecHitFromSoA", "EcalUncalibRecHitsEE"),
+    # RecHit GPU collection tags are temp placeholder
+    EBCpuRecHit = cms.untracked.InputTag("hltEcalRecHitWithoutTPs", "EcalRecHitsEB"),
+    EECpuRecHit = cms.untracked.InputTag("hltEcalRecHitWithoutTPs", "EcalRecHitsEE"),
+    EBGpuRecHit = cms.untracked.InputTag("hltEcalRecHitWithTPs", "EcalRecHitsEB"),
+    EEGpuRecHit = cms.untracked.InputTag("hltEcalRecHitWithTPs", "EcalRecHitsEE")
```

Comment:
I'm going to follow a suggestion from DQMDC to have the online GPU module in its own dedicated client like HCAL did here: https://github.com/cms-sw/cmssw/blob/master/DQM/Integration/python/clients/hcalgpu_dqm_sourceclient-live_cfg.py. This way we can keep the offline values here and modify them for online in the new client, and we can also avoid crashing the standard ECAL client in case of issues with the GPU stream.

Reply:
Yes, that makes sense.

In fact, the online client for the GPU vs CPU validation _has to_ be separate from the regular one, because the HLT collections come in a different stream than the raw data.